splice_machine/src/main/java/com/splicemachine/backup/SnapshotSystemProcedure.java

```diff
@@ -0,0 +1,107 @@
+import com.splicemachine.procedures.ProcedureUtils;
+import com.splicemachine.snapshot.SnapshotManager;
+import com.splicemachine.utils.SpliceLogUtils;
+import org.apache.log4j.Logger;
+import java.sql.ResultSet;
+import java.util.List;
+
+public class SnapshotSystemProcedure {
+
+    private static Logger LOG = Logger.getLogger(SnapshotSystemProcedure.class);
+
+    public static void addProcedures(List<Procedure> procedures) {
+        Procedure snapshotSchema = Procedure.newBuilder().name("CREATE_SCHEMA_SNAPSHOT")
+                .varchar("schemaName", 128)
+                .sqlControl(RoutineAliasInfo.MODIFIES_SQL_DATA)
+                .numOutputParams(0)
+                .numResultSets(1)
+                .ownerClass(SnapshotSystemProcedure.class.getCanonicalName())
+                .build();
+        procedures.add(snapshotSchema);
+
+        Procedure cloneSnapshot = Procedure.newBuilder().name("CLONE_SCHEMA_SNAPSHOT")
+                .bigint("snapshotId")
+                .catalog("schemaName")
+                .sqlControl(RoutineAliasInfo.MODIFIES_SQL_DATA)
+                .numOutputParams(0)
+                .numResultSets(1)
+                .ownerClass(SnapshotSystemProcedure.class.getCanonicalName())
+                .build();
+        procedures.add(cloneSnapshot);
+
+        Procedure deleteSnapshot = Procedure.newBuilder().name("DELETE_SNAPSHOT")
+                .bigint("snapshoId")
+                .numOutputParams(0)
+                .numResultSets(1)
+                .ownerClass(SnapshotSystemProcedure.class.getCanonicalName())
+                .build();
+        procedures.add(deleteSnapshot);
+    }
+    public static void CLONE_SCHEMA_SNAPSHOT(long snapshotId, String schemaName, ResultSet[] resultSets) throws Exception {
+        try {
+            SnapshotManager snapshotManager = EngineDriver.driver().manager().getSnapshotManager();
+            snapshotManager.cloneSchemaSnapshot(snapshotId, schemaName);
+        }catch (Throwable t) {
+            resultSets[0] = ProcedureUtils.generateResult("Error", t.getLocalizedMessage());
+            SpliceLogUtils.error(LOG, "Database backup error", t);
+            t.printStackTrace();
+        }
+    }
+
+    /**
+     * Take a snapshot of a schema
+     * @param schemaName
+     * @throws Exception
+     */
+    public static void CREATE_SCHEMA_SNAPSHOT(String schemaName, ResultSet[] resultSets) throws Exception {
+        try {
+            schemaName = EngineUtils.validateSchema(schemaName);
+            EngineUtils.checkSchemaVisibility(schemaName);
+
+            SnapshotManager snapshotManager = EngineDriver.driver().manager().getSnapshotManager();
+            snapshotManager.createSchemaSnapshot(schemaName);
+        }catch (Throwable t) {
+            resultSets[0] = ProcedureUtils.generateResult("Error", t.getLocalizedMessage());
+            SpliceLogUtils.error(LOG, "Database backup error", t);
+            t.printStackTrace();
+        }
+    }
```

Comment:
i think CREATE_SCHEMA_SNAPSHOT should return some information about the snapshot, at least the snapshot id. if we don't return the snapshot id, you have to always additionally do a `select * from sys.syssnapshotitems`, then get the latest created (+ hope noone created one in the meantime or check also schema etc). At best, it would just return the new row created sys.syssnapshotitems.
Btw, can we somehow have information about how much memory a snapshot consumes in `sys.syssnapshotitems`? admins might want to clean large snapshots first if they get low on hdfs memory.

Reply:
I will file some Jiras to followup your comments

Reply:
Filed DB-12240 and DB-12241